### PR TITLE
Add distribution test for OpenID and fix issues with jetty-openid.xml

### PR DESCRIPTION
**Pull request by lachlan-roberts**
## Issue #7545

- Add `@Name` annotations to `OpenIdConfiguration` constructor.
- Add distribution test for jetty-openid so that any issues with the XML are detected.

### CODE REVIEW

File: jetty-openid/src/main/java/org/eclipse/jetty/security/openid/OpenIdConfiguration.java [hunk 1]
```diff
@@ -86,8 +87,10 @@ public OpenIdConfiguration(String issuer, String authorizationEndpoint, String t
      * @param authMethod Authentication method to use with the Token Endpoint.
      * @param httpClient The {@link HttpClient} instance to use.
      */
-    public OpenIdConfiguration(String issuer, String authorizationEndpoint, String tokenEndpoint,
-                               String clientId, String clientSecret, String authMethod, HttpClient httpClient)
+    public OpenIdConfiguration(@Name("issuer") String issuer,
+                               @Name("authorizationEndpoint") String authorizationEndpoint, @Name("tokenEndpoint") String tokenEndpoint,
+                               @Name("clientId") String clientId, @Name("clientSecret") String clientSecret,
+                               @Name("authMethod") String authMethod, @Name("httpClient") HttpClient httpClient)
```
Comment (gregw) [inline jetty-openid/src/main/java/org/eclipse/jetty/security/openid/OpenIdConfiguration.java]:
```suggestion
    public OpenIdConfiguration(@Name("issuer") String issuer,
                               @Name("authorizationEndpoint") String authorizationEndpoint,
                               @Name("tokenEndpoint") String tokenEndpoint,
                               @Name("clientId") String clientId,
                               @Name("clientSecret") String clientSecret,
                               @Name("authMethod") String authMethod,
                               @Name("httpClient") HttpClient httpClient)
```

File: tests/test-webapps/test-openid-webapp/src/main/java/org/eclipse/jetty/test/openid/ErrorPage.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+public class ErrorPage extends HttpServlet
+{
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException
+    {
+        response.setContentType("text/html");
+        response.getWriter().println("not authorized");
+        response.getWriter().println("<br><a href=\"" + request.getContextPath() + "\">Home</a>");
+    }
+}
```
Comment (gregw) [inline tests/test-webapps/test-openid-webapp/src/main/java/org/eclipse/jetty/test/openid/ErrorPage.java]: EOLN

File: tests/test-webapps/test-openid-webapp/src/main/java/org/eclipse/jetty/test/openid/HomePage.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            response.getWriter().println("email: " + userInfo.get("email") + "<br>");
+            response.getWriter().println("<br><a href=\"" + request.getContextPath() + "/logout\">Logout</a>");
+        }
+        else
+        {
+            response.getWriter().println("not authenticated");
+            response.getWriter().println("<br><a href=\"" + request.getContextPath() + "/login\">Login</a>");
+        }
+    }
+}
```
Comment (gregw) [inline tests/test-webapps/test-openid-webapp/src/main/java/org/eclipse/jetty/test/openid/HomePage.java]: EoLN

File: tests/test-webapps/test-openid-webapp/src/main/java/org/eclipse/jetty/test/openid/LoginPage.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+public class LoginPage extends HttpServlet
+{
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException
+    {
+        response.setContentType("text/html");
+        response.getWriter().println("success");
+        response.getWriter().println("<br><a href=\"" + request.getContextPath() + "\">Home</a>");
+    }
+}
```
Comment (gregw) [inline tests/test-webapps/test-openid-webapp/src/main/java/org/eclipse/jetty/test/openid/LoginPage.java]: EOLN

File: tests/test-webapps/test-openid-webapp/src/main/java/org/eclipse/jetty/test/openid/LogoutPage.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+public class LogoutPage extends HttpServlet
+{
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException
+    {
+        request.getSession().invalidate();
+        response.sendRedirect(request.getContextPath());
+    }
+}
```
Comment (gregw) [inline tests/test-webapps/test-openid-webapp/src/main/java/org/eclipse/jetty/test/openid/LogoutPage.java]: EOLN

**Review by gregw — DISMISSED** [review]
some niggles

**Review by olamy — APPROVED** [review]

**Pull request merged**